controllers/limitador_controller.go

```diff
@@ -109,24 +113,111 @@ func (r *LimitadorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *LimitadorReconciler) reconcileStatus(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) error {
+func (r *LimitadorReconciler) reconcileStatus(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) (err error) {
 	logger := logr.FromContext(ctx)
-	// Simple enough now, we could implement a thorough check like with RateLimit with ObservedGeneration in the future
-	builtServiceUrl := buildServiceUrl(limitadorObj)
-	if builtServiceUrl != limitadorObj.Status.ServiceURL {
-		logger.V(1).Info("Updating the Status", "Name", limitadorObj.Name)
-		limitadorObj.Status.ServiceURL = builtServiceUrl
-		return r.Client().Status().Update(ctx, limitadorObj)
 
+	isLimitadorRunning := r.checkLimitadorInstanceIsRunning(ctx, limitadorObj)
+	changed := updateStatusReady(limitadorObj, isLimitadorRunning)
+
+	changed = updateStatusService(limitadorObj) || changed
+
+	if !limitadorObj.Status.Ready() {
+		err = fmt.Errorf("resource not ready")
+	}
+
+	if !changed {
+		logger.V(1).Info("resource status did not change")
+		return // to save an update request
+	}
+
+	logger.V(1).Info("resource status changed", "limitador/status", limitadorObj.Status)
+
+	if updateErr := r.Client().Status().Update(ctx, limitadorObj); updateErr != nil {
+		logger.Error(updateErr, "failed to update the resource")
+		err = updateErr
+		return
+	}
+
+	logger.Info("status updated", "Name", limitadorObj.Name)
+	return
+}
+
+func (r *LimitadorReconciler) checkLimitadorInstanceIsRunning(ctx context.Context, limitadorObj *limitadorv1alpha1.Limitador) bool {
+	logger := logr.FromContext(ctx)
+	limitadorInstance := &appsv1.Deployment{}
+	limitadorInstanceNamespacedName := client.ObjectKey{ // Its deployment is built after the same name and namespace
+		Namespace: limitadorObj.Namespace,
+		Name:      limitadorObj.Name,
 	}
-	return nil
+	if err := r.Client().Get(ctx, limitadorInstanceNamespacedName, limitadorInstance); err != nil {
```

Comment:
if not found -> return false
it other error -> return error

Reply:
Makes sense, although I'll need to refactor the logic of that particular function and how it's treated on the reconciling flow. Related to this: https://github.com/Kuadrant/limitador-operator/pull/33#discussion_r927544384